### PR TITLE
Fixes airflow workflow registration bugs

### PR DESCRIPTION
**Pull request by saurav-c**
## Describe your changes and why you are making these changes
This PR fixes a couple bugs for registering airflow workflows. These came up when recording a demo. Comments are provided inline for why these bug fixes are needed and if there is any follow on work.

## Related issue number (if any)
ENG 2600

## Loom demo (if any)
This is also how I performed testing for these changes. Please see https://www.loom.com/share/ffe3fa8086e244e5ad06faac87d6630c

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/cmd/server/handler/register_airflow_workflow.go [hunk 1]
```diff
@@ -92,7 +92,7 @@ func (h *RegisterAirflowWorkflowHandler) Prepare(r *http.Request) (interface{},
 		dagSummary.Dag.Metadata.Name,
 		h.Database,
 	)
-	if err != nil {
+	if err != nil && !errors.Is(err, database.ErrNoRows()) {
```
Comment (saurav-c) [inline src/golang/cmd/server/handler/register_airflow_workflow.go:95]: We were missing this critical check of the error type, since `database.ErrNoRows` is actually what we want

File: src/golang/lib/workflow/operator/operator.go [hunk 2]
```diff
@@ -129,7 +129,10 @@ func NewOperator(
 	var jobManager job.JobManager
 	var err error
 
-	if dagJobManager == nil {
+	if dagJobManager == nil && opEngineConfig.Type != shared.AirflowEngineType {
```
Comment (saurav-c) [inline src/golang/lib/workflow/operator/operator.go:132]: As commented, we do not need a job manager for Airflow operators, since there is job being deployed

File: src/python/aqueduct_executor/operators/airflow/dag.template [hunk 3]
```diff
@@ -11,7 +11,6 @@ VENV_REQUIREMENTS=[
     "cloudpickle",
     "distro",
     "google-cloud-bigquery",
-    "mysqlclient",
```
Comment (saurav-c) [inline src/python/aqueduct_executor/operators/airflow/dag.template:14]: See https://linear.app/aqueducthq/issue/ENG-2602/add-support-for-using-mysql-with-airflow for why this was causing issues. 

tl;dr we cant install mysqlclient by default

**Review by cw75 — APPROVED** [review]
